src/connectStyle.js

```diff
@@ -272,10 +259,27 @@ export default function connectStyle(
       }
 
       render() {
-        const { addedProps, style } = this.state;
+        const { virtual, parentStyle, children, ...otherProps } = this.props;
+        const { addedProps, style, childrenStyle } = this.state;
+
+        const resolvedParentStyle = virtual ? parentStyle : childrenStyle;
+
+        const childrenWithProps = Children.map(children, child => {
+          if (isValidElement(child)) {
+            return cloneElement(child, { parentStyle: resolvedParentStyle, transformProps: this.transformProps })
+          }
+
+          return child;
+        });
```

Comment:
Could you please elaborate on this, I'd love to understand this part.

Why do we apply props only to valid elements and what is an invalid element?

Why do we return invalid elements without props being added?